### ISSUE TITLE
[HTTP API] URLs with underscores in the domain name are rejected as malformed

### ISSUE THREAD

**Issue by Kimapr** [cc-tweaked/CC-Tweaked#959]
### Minecraft Version

1.16.x

### Version

1.98.2

### Details

In my setup i use hostnames in /etc/hosts with underscores in them, and CC rejects URLs with them even though they are perfectly valid.

    lua> http.get("http://kdp_portable/")
    nil
    "URL malformed"

**Comment by SquidDev** [comment]
Right, this is because `kdp_portable` is an invalid hostname - they can only contain alpha-numeric characters and `.` (see [RFC 952]).

```
jshell> import java.net.URI;
jshell> var uri = new URI("http://kdp_portable/")
uri ==> http://kdp_portable/
jshell> uri.getHost()
$3 ==> null
jshell> uri.getAuthority()
$4 ==> "kdp_portable"
```

I guess what one could do is parse the authority, strip out the user and port information. Though Not Ideal, and something I'd rather avoid doing.

[RFC 952]: https://datatracker.ietf.org/doc/html/rfc952

**Comment by Kimapr** [comment]
> Right, this is because kdp_portable is an invalid hostname - they can only contain alpha-numeric characters and . (see RFC 952).

But it works in any other application, so why won't it work in computercraft too?

**Comment by SquidDev** [comment]
Because nobody follows internet standards very thoroughly. Apart from the Java stdlib apparently.
